### PR TITLE
Feature/ntgl 820 create friend group modal

### LINKED ISSUE #820: feat: Create friend group modal

### PR DESCRIPTION
**Pull request by jasttran**
Created a new component 'AddGroupButton' that appears in the Navbar and opens a dialogue to create a new friend group.

It is composed of:
- **StyledDialogTitle** which contains the 'Add a Group' title and the 'X' close button
- **StyledUploadImageContainer** which is composed of the two circles 
    - **CircleOutline**: displays the image
    - **EditImagePopOver**: edit icon that opens a popover with a Textfield to input image address.
- **Group Name Text Field**
- **Select Group Members Autocomplete Component** where the properties _renderTags_ is the customization of the selected member chips, and _renderOption_ is the customization of the options listed in the dropdown.
- **StyledDialogActions** which contains the Cancel and Create buttons at the bottom of the dialog.
image

- [ ] Remove the dummy friends database at the bottom of the file
- [ ] Once backend is completed, implement creating a group
- [ ] Default image

### CODE REVIEW

**Comment by Rayahhhmed** [comment]
nice draft pr

File: client/src/components/sidebar/addGroupDialog/AddGroupDialogContent.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  object-fit: cover;
+`;
+
+const StyledUploadImageContainer = styled('div')`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-end;
+`;
+
+interface AddGroupDialoContentProps {
```
Comment (Rayahhhmed) [inline client/src/components/sidebar/addGroupDialog/AddGroupDialogContent.tsx]: nit: ``AddGroupDialogContentProps``

**Comment by jasttran** [comment]
New and better branch

**Comment by mark-trann** [comment]
This was moved to other branch and merged to dev already: https://github.com/devsoc-unsw/notangles/pull/882

**Pull request closed without merging**